src/Storages/ObjectStorage/S3/Configuration.cpp

```diff
@@ -345,10 +351,60 @@ void StorageS3Configuration::fromAST(ASTs & args, ContextPtr context, bool with_
             engine_args_to_idx = {{"access_key_id", 1}, {"secret_access_key", 2}, {"session_token", 3}, {"format", 4}, {"compression_method", 5}};
         }
     }
+    /// For 7 arguments we support:
```

Comment:
Let's not forget to update documentation

Reply:
I'll update it once we get closer to approval & merge if that's ok